### ISSUE TITLE
Cannot access triples outside the default graph / handling partitioned graphs

### ISSUE THREAD

**Issue by cbobed** [Rothamsted/rdf2neo#1]
I've managed to make the whole system work, but it seems that it cannot handle a tdb which is partitioned into graphs. 

I've digged a little bit into the code and I've reached SparqlEndPointHelper and TDBEndPointHelper (@rdfutils project), where, to execute the queries (e.g., processSelect), it gets a queryExecutor via getQueryExecutor(). This method is abstract at SparqlEndPointHelper, and it is implemented using this.getDefaultModel() @ TDBEndpoint. Is it the intended behaviour? 

Apart from that, in the RDFManager there are several points where getDefaultModel() is used. I've changed it to getUnionModel() and also using directly Dataset; just in case it is helpful.

**Comment by marco-brandizi** [comment]
Hi @cbobed, thank you for your interest in our project! My initial idea was to use the default model and let the user decide to use other named graphs by specifying them in the configured SPARQL queries. The reason for this is that one might have the actual data in the default and support data (eg, metadata, VoID) in other graphs, so using only the default model when nothing else is specified should make the query execution a bit faster (hoping the engine is able to efficiently exclude all other graphs). I think we should support the union as an alternative configurable option. I will work on such an extension, which, if we want it done properly, will require a Spring qualifier in order to support configurability (the same way it's done in methods like, `RdfDataManager.setCyRelation
TypeIdConverter ()`). Should you want to contribute, pull requests are welcome! :-)

**Comment by cbobed** [comment]
Hi, first of all, congrats for your work ! :) 

I've come with a relatively simpler workaround: 
1st) in the rdfutils project, the TDBEndPoint: 
getQueryExecutor: 
//		Model m = this.getDataSet ().getDefaultModel ();
		return QueryExecutionFactory.create ( query, this.getDataSet() );

In this way, the queries are executed against the complete dataset. Otherwise, eventhough the user posed a query with a GRAPH clause, as it relied on super.processSelect(), it ended up posing the query to the defaultModel(). 

2n) in the rdf2neo project, RDFManager has to be slighlty updated to use the UnionModel and the dataset instead of the default model. 

This would get rid of the need of further configuration, as everything could be done via the SPARQL queries. I've already used it with a GRAPH ?x clause and it works both with triples in the defaultGraph and the rest of the knowledge graph. 

If you guide me (and you consider the solution worthy), I would gladly submit a pull request :)

**Comment by marco-brandizi** [comment]
Thanks! I'll come back to this. I believe this is an edge case, so it should be enough to use the union model and to let the users to possibly add `FROM NAMED <>` to their SPARQL queries. So, if we make the union model (or `getDataSet()`) hardwired (no configurable option, contrary to what I wrote earlier), you already have written the solution in the comments :-) If you've time, you can make a fork of this repo change your fork and then [send us a pull request](https://help.github.com/en/articles/creating-a-pull-request).

**Comment by cbobed** [comment]
I will do it as soon as possible :) 

By the way, I'm getting duplicated triples coming from different queries: I've double checked that it is exactly the same iri for both triples, but I get the edges duplicated in the property graph (the inner IDs are different). 

I've got a test with classes Person and Location (among others). Person has a property with range Location, and I've put it in the Person_rel_type sparql query. I've done the same for Location (the domain of hasLocation is owl:Thing), but I got one different edge for each query execution in spite of having exactly the same iri (I've used an artificial "namespace" built concatenating the URI of the property + "/" + the MD5 hash of the triple). Is it normal? I thought I was safe from duplications using the BIND clause you use for the iri generation. 

Besides, although I'm pretty sure it is commented in the paper, what does it happen exactly to non-seen nodes? are they built on-the-fly?
